core/container/src/main/java/org/wildfly/swarm/Swarm.java

```diff
@@ -464,7 +464,13 @@ private static boolean isFatJar() throws IOException {
 
     private void loadStageConfiguration(URL url) {
         List<ProjectStage> projectStages = new ProjectStageFactory().loadStages(url);
-        String stageName = System.getProperty(SwarmProperties.PROJECT_STAGE, "default");
+        String stageName = System.getProperty(SwarmProperties.PROJECT_STAGE);
+        if (stageName == null) {
+            stageName = System.getenv(SwarmProperties.PROJECT_STAGE);
```

Comment:
In this case environment variable must be called `swarm.project.stage`, what I would do is a `PROJECT_STAGE.replace('.', '_')` to be more environment fashion.